src/components/textarea/textarea.jsx

```diff
@@ -0,0 +1,129 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useRef,
+  useImperativeHandle,
+} from 'react';
+import cautionSvg from '../../svgs/caution.svg';
+import FormControl from '../form-control/form-control.jsx';
+import Icon from '../icon/icon.jsx';
+import useMounted from '../../hooks/use-mounted.js';
+import useValidation from '../../hooks/use-validation.jsx';
+import styles from './textarea.css';
+
+function getClassName(className, validationMessage) {
+  if (className) return className;
+  return validationMessage ? styles['invalid-textarea'] : styles.textarea;
+}
+
+const Textarea = forwardRef(function Textarea({
```

Comment:
Outta curiosity, do you prefer named (destructured) props instead of `props` parameter? Or are ya feeling things out?

Reply:
Feeling it out. Honestly it doesn't feel too bad. I'm worried about variable collisions, but didn't run into that here, and some extra lines in exchange for not needing `props.` felt nice.